docs/topics/request-response.rst

```diff
@@ -200,12 +214,37 @@ Example::
         self.logger.info("Visited %s", response.url)
 
 In some cases you may be interested in passing arguments to those callback
-functions so you can receive the arguments later, in the second callback. You
-can use the :attr:`Request.meta` attribute for that.
+functions so you can receive the arguments later, in the second callback.
+The following example shows how to achieve this by using the
+:attr:`Request.cb_kwargs` attribute:
+
+::
+
+    # pass information to the next callback using the Request.cb_kwargs attribute
+    def parse(self, response):
+        request = scrapy.Request('http://www.example.com/index.html',
+                                 callback=self.parse_page2,
+                                 cb_kwargs=dict(main_url=response.url))
+        request.cb_kwargs['foo'] = 'bar'  # add more arguments for the callback
+        yield request
+
+    def parse_page2(self, response, main_url, foo):
+        yield dict(
+            main_url=main_url,
+            other_url=response.url,
+            foo=foo,
+        )
 
-Here's an example of how to pass an item using this mechanism, to populate
-different fields from different pages::
+.. caution:: :attr:`Request.cb_kwargs` was introduced in version ``1.7``.
+   Prior to that, :attr:`Request.meta` was the recommended option for passing
+   information around callbacks. However, after ``1.7``, using :attr:`Request.cb_kwargs`
+   became the preferred way of passing user information, leaving :attr:`Request.meta`
+   to be populated by internal components like spider or downloader middlewares.
```

Comment:
> leaving :attr:`Request.meta` to be populated by internal components like spider or downloader middlewares

I think this part is a bit misleading. It sounds like users will never use meta from a spider. It should sound like they are going to write and read meta fields only to exchange information with middlewares and extensions. And I would not call those components internal, since users are expected to write them themselves.

Of course, this does not matter if we remove the outdated documentation altogether.

Reply:
:ok_hand: